common/models/acl.js

```diff
@@ -456,15 +489,34 @@ module.exports = function(ACL) {
         if (err) return callback(err, null);
 
         var resolved = self.resolvePermission(effectiveACLs, req);
+        var permission = resolved && resolved.permission;
         if (resolved && resolved.permission === ACL.DEFAULT) {
-          resolved.permission = (model && model.settings.defaultPermission) || ACL.ALLOW;
+          permission = defaultPermission || ACL.ALLOW;
         }
```

Comment:
you were proposing to share here the same logic as in ACL.isAllowed() but here were are not quite doing the same thing and not testing the same type of object: 
1. resolved is not an instance of ACL but an instance of AccessRequest. I'm reluctant in using ACL.isAllowed() on the permission part of an AccessRequest instance. Proper way would be to use the existing [AccessRequest.prototype.isAllowed](https://github.com/strongloop/loopback/blob/master/lib/access-context.js#L317) after adapting it to accept a default permission
2. we are not testing if allowed, we are changing the permission type if actually DEFAULT, because it is not done either [here](https://github.com/strongloop/loopback/blob/master/common/models/acl.js#L494) or [there](https://github.com/strongloop/loopback/blob/master/lib/access-context.js#L318) which is used in [model.js](https://github.com/strongloop/loopback/blob/master/lib/model.js#L353). This could be avoided and the code simplified/unified if were to align the implementation of `AccessRequest.prototype.isAllowed`

If you agree, i can track this as a different issue and address later